src/resolve/resolved-dns-answer.c

```diff
@@ -54,12 +92,20 @@ static int dns_answer_add_raw(DnsAnswer *a, DnsResourceRecord *rr, int ifindex,
         if (a->n_rrs >= a->n_allocated)
                 return -ENOSPC;
 
-        a->items[a->n_rrs++] = (DnsAnswerItem) {
+        a->items[a->n_rrs] = (DnsAnswerItem) {
                 .rr = dns_resource_record_ref(rr),
                 .ifindex = ifindex,
                 .flags = flags,
         };
 
+        r = set_put(a->set_items, &a->items[a->n_rrs]);
+        if (r < 0)
+                return r;
+        if (r == 0)
+                return -EEXIST;
```

Comment:
a->items[a->n_rrs].rr has been ref'ed above, what happens to it if we return early?